app/classifier/frame-annotator.jsx

```diff
@@ -0,0 +1,282 @@
+    const createdViewBox = `${this.props.viewBoxDimensions.x} ${this.props.viewBoxDimensions.y} ${this.props.viewBoxDimensions.width} ${this.props.viewBoxDimensions.height}`;
+
+    if (this.props.annotation) {
+      taskDescription = this.props.workflow.tasks[this.props.annotation.task];
+      TaskComponent = tasks[taskDescription.type];
+    }
+
+    if (this.state.alreadySeen) {
+      warningBanner = (
+        <WarningBanner label="Already seen!">
+          <p>Our records show that you’ve already seen this image. We might have run out of data for you in this workflow!</p>
+          <p>Try choosing a different workflow or contributing to a different project.</p>
+        </WarningBanner>
+      );
+    } else if (this.props.subject.retired) {
+      warningBanner = (
+        <WarningBanner label="Finished!">
+          <p>This subject already has enough classifications, so yours won’t be used in its analysis!</p>
+          <p>If you’re looking to help, try choosing a different workflow or contributing to a different project.</p>
+        </WarningBanner>
+      );
+    }
+
+    const svgStyle = {};
+    if (type === 'image' && !this.props.loading) {
+      // Images are rendered again within the SVG itself.
+      // When cropped right next to the edge of the image,
+      // the original tag can show through, so fill the SVG to cover it.
+      svgStyle.background = 'black';
+
+      // Allow touch scrolling on subject for mobile and tablets
+      if (taskDescription) {
+        if ((taskDescription.type !== 'drawing') && (taskDescription.type !== 'crop')) {
+          svgStyle.pointerEvents = 'none';
+        }
+      }
+      if (this.props.panEnabled === true) {
+        svgStyle.pointerEvents = 'all';
+      }
+    }
+
+    const svgProps = {};
+
+    if (TaskComponent) {
+      BeforeSubject = TaskComponent.BeforeSubject;
+      InsideSubject = TaskComponent.InsideSubject;
+      AfterSubject = TaskComponent.AfterSubject;
+    }
+
+    const hookProps = {
+      taskTypes: tasks,
+      workflow: this.props.workflow,
+      task: taskDescription,
+      classification: this.props.classification,
+      annotation: this.props.annotation,
+      frame: this.props.frame,
+      scale: this.getScale(),
+      naturalWidth: this.props.naturalWidth,
+      naturalHeight: this.props.naturalHeight,
+      containerRect: this.getSizeRect(),
+      getEventOffset: this.getEventOffset,
+      onChange: this.props.onChange,
+      preferences: this.props.preferences,
+      normalizeDifference: this.normalizeDifference,
+      getScreenCurrentTransformationMatrix: this.getScreenCurrentTransformationMatrix
+    };
+
+    Object.keys(tasks).map((task) => {
+      const Component = tasks[task];
+      if (Component.getSVGProps) {
+        const props = Component.getSVGProps(hookProps);
+        if (props) {
+          Object.keys(props).map((key) => {
+            svgProps[key] = props[key];
+          });
+        }
```

Comment:
I'm not 100% sure what this is doing, but I think `Object.assign(svgProps, props)` would work, or maybe even `Object.assign(svgProps, Component.getSVGProps(hookProps));`